plugins/UsersManager/API.php

```diff
@@ -418,14 +478,95 @@ public function getSitesAccessFromUser($userLogin)
                     'site' => $site['idsite'],
                     'access' => 'admin'
                 );
-
             }
             return $return;
         }
-
         return $this->model->getSitesAccessFromUser($userLogin);
     }
 
+    /**
+     * For each website ID, returns the access level of the given $userLogin (if the user is not a superuser).
+     * If the user doesn't have any access to a website ('noaccess'),
+     * this website will not be in the returned array.
+     * If the user doesn't have any access, the returned array will be an empty array.
+     *
+     * @param string $userLogin User that has to be valid
+     *
+     * @param int|null $limit
+     * @param int|null $offset
+     * @param string|null $filter_search text to search for in site name, URLs, or group.
+     * @param string|null $filter_access access level to select for, can be 'some', 'view' or 'admin' (by default 'some')
+     * @return array    The returned array has the format
+     *                    array(
+     *                        ['idsite' => 1, 'site_name' => 'the site', 'access' => 'admin'],
+     *                        ['idsite' => 2, 'site_name' => 'the other site', 'access' => 'view'],
+     *                        ...
+     *                    )
+     * @throws Exception
+     */
+    public function getSitesAccessForUser($userLogin, $limit = null, $offset = 0, $filter_search = null, $filter_access = null)
+    {
+        Piwik::checkUserHasSomeAdminAccess();
+        $this->checkUserExists($userLogin);
+
+        if (Piwik::hasTheUserSuperUserAccess($userLogin)) {
+            throw new \Exception("This method should not be used with superusers.");
+        }
+
+        $idSites = null;
+        if (!Piwik::hasUserSuperUserAccess()) {
+            $idSites = $this->access->getSitesIdWithAdminAccess();
```

Comment:
just as a "safety net". Can we throw an exception if `idSites` is empty and doesn't return any site? I know it shouldn't as user has at least admin access... just to be double safe maybe.

Reply:
That should already be checked by `Piwik::checkUserHasSomeAdminAccess();` above.

Reply:
But I'll add a sanity check.